src/nuntius_mocker.erl

```diff
@@ -118,15 +118,20 @@ run_expects(Message, Expects) ->
 
 mfa_from_fun_info(Expect) ->
     {module, M} = erlang:fun_info(Expect, module),
-    {name, F} = erlang:fun_info(Expect, name),
+    {name, F0} = erlang:fun_info(Expect, name),
+    F = coerce_fun(F0),
     {arity, A} = erlang:fun_info(Expect, arity),
     {M, F, A}.
 
 mfa_from_stacktrace(Stacktrace) ->
-    [{M, F, Args, _} | _] = Stacktrace,
+    [{M, F0, Args, _} | _] = Stacktrace,
     A = length(Args),
+    F = coerce_fun(F0),
     {M, F, A}.
```

Comment:
Not sure if we know of a better, more modern way of doing this, actually. It's served our purposes until now 😄

Reply:
I'm a little bit skeptical of coercing the fun name in `mfa_from_fun_info/1` since we do want to match only against the specific expectation function head there… not any of its internal ones… nor any external one. I guess tests were failing in OTP26 and that triggered this adjustments, right?

Reply:
Correct. A function head that should not throw will look like

```erlang
[
{nuntius_api_SUITE,'-expect_message/1-inlined-0-',[unknown],[
  {file,"nuntius/test/nuntius_api_SUITE.erl"},{line,206}
]},
{nuntius_mocker,'-run_expects/2-fun-0-',4,[
  {file,"nuntius/src/nuntius_mocker.erl"},{line,105}
]},
{maps,fold_1,3,[
  {file,"maps.erl"},{line,411}
]},
{nuntius_mocker,handle_message,2,[
  {file,"nuntius/src/nuntius_mocker.erl"},{line,97}
]},
{nuntius_mocker,loop,1,[
  {file,"nuntius/src/nuntius_mocker.erl"},{line,71}
]},
{proc_lib,init_p_do_apply,3,[
  {file,"proc_lib.erl"},{line,240}
]}
]
```

where StMFA is

```erlang
{nuntius_api_SUITE,'-expect_message/1-inlined-0-',1}
```

and FiMFA is

```erlang
{nuntius_api_SUITE,'-expect_message/1-fun-0-',1}
```

meaning the call is in `nuntius_api_SUITE:206` where we're aiming for a call no expect is able to handle.

Reply:
> I doubt that OTP provides a way of determining if some function "is written" inside another one

At the time we created this, I don't think it did. And this was something I actually discussed with @eproxus and a hint for the solution was born from that discussion.

Reply:
I'm trying to improve this, but:

* should throw is
```erlang
[
{lists,sort,[fff],[]},
{nuntius_mocker,'-run_expects/2-fun-0-',4,[]},
{maps,fold_1,3,[]},
{nuntius_mocker,handle_message,2,[]},
{nuntius_mocker,loop,1,[]},
{proc_lib,init_p_do_apply,3,[]}
]
```
* should not throw is
```
{st,[
{nuntius_api_SUITE,'-expect_message/1-inlined-0-',[unknown],[]},
{nuntius_mocker,'-run_expects/2-fun-0-',4,[]},
{maps,fold_1,3,[]},
{nuntius_mocker,handle_message,2,[]},
{nuntius_mocker,loop,1,[]},
{proc_lib,init_p_do_apply,3,[]}
]}
```

so I'm out of ideas on what to search (the head does contain the information we're looking for, though, since we're just matching the exception on function clauses - don't know if you saw/remember that bit)

Reply:
Resolve'ing for readability. This is dealt with further, as per discussion.